examples/update-on-async-rendering/updating-state-from-props-after.js

```diff
@@ -0,0 +1,29 @@
+// After
+class ExampleComponent extends React.Component {
+  // highlight-range{1-3}
+  // Initialize state in constructor,
+  // Or with a property initializer.
+  state = {};
+
+  // highlight-range{1-20}
+  static getDerivedStateFromProps(
+    nextProps,
+    prevState
+  ) {
+    if (
+      prevState.someMirroredValue !==
+      nextProps.someValue
+    ) {
+      return {
+        derivedData: computeDerivedState(
+          nextProps
+        ),
+        someMirroredValue:
```

Comment:
I think especially here this is getting so abstract it's hard to see what's going on. (What is mirroring? what is derived? what you put where and why?)